### PR TITLE
Fix reading hitmap on every mouse move

### PR DESCRIPTION
**Pull request by davidswinegar**
## Summary

We had some incorrect logic that would cause the Worldview component
to read from the hitmap on every mouse move even if hitmapOnMouseMove
was false, as long as there was no `onMouseMove` handler. Fix it and
make the logic path a bit clearer.

## Test plan

Test manually with interactions on the demo page to make sure they aren't
broken.

## Versioning impact

Requires a patched version.

### CODE REVIEW

File: packages/regl-worldview/src/Worldview.js [hunk 1]
```diff
@@ -196,8 +196,12 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
       return;
     }
 
-    if (!readHitmap && worldviewHandler) {
-      return handleWorldviewMouseInteraction(0, ray, e, worldviewHandler);
+    // rendering the hitmap on mouse move is expensive, so disable it by default
+    if (mouseEventName === "onMouseMove" && !this.props.hitmapOnMouseMove) {
```
Comment (esthersweon) [inline packages/regl-worldview/src/Worldview.js]: Could you pull out into variable like `const renderHitMapOnMouseMove = mouseEventName === "onMouseMove" && this.props.hitmapOnMouseMove` and say `if (!renderHitmapOnMouseMove)`? Would that do what you want it to do?

File: packages/regl-worldview/src/Worldview.js [hunk 2]
```diff
@@ -196,8 +196,12 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
...
+      if (worldviewHandler) {
+        return handleWorldviewMouseInteraction(0, ray, e, worldviewHandler);
+      }
+      return;
```
Comment (esthersweon) [inline packages/regl-worldview/src/Worldview.js]: #nit for clearer logic path: 
```
if (!this.props.hitmapOnMouseMove) {
  if (mouseEventName === "onMouseMove") {
    return;
  }
  if (worldviewHandler) {
    return handleWorldviewMouseInteraction(0, ray, e, worldviewHandler);
  }
}

Comment (davidswinegar) [reply packages/regl-worldview/src/Worldview.js]: I don't think this is exactly the same logic. I could change it to:
```
    if (mouseEventName === "onMouseMove" && !this.props.hitmapOnMouseMove) {
      if (worldviewHandler) {
        return handleWorldviewMouseInteraction(0, ray, e, worldviewHandler);
      } else {
        return;
      }
    }
```
Would that help?

Comment (robin-pham) [reply packages/regl-worldview/src/Worldview.js]: think the original is fine, this `else` is extraneous

**Pull request merged**